CONTRIBUTING.md

```diff
@@ -0,0 +1,98 @@
+# Contributing
+
+## Found a Bug?
+
+If you find a bug in the source code, you can help us by submitting an issue to our GitHub Repository. Even better, you can submit a Pull Request with a fix.
+
+## Submit an Issue
+
+Please fill the following information in each issue you submit:
+
+* Title: Use a clear and descriptive title for the issue to identify the problem.
+* Description: Description of the issue.
+* Steps to Reproduce: numbered step by step. (1,2,3.… and so on)
+* Expected behaviour: What you expect to happen.
+* Actual behaviour: What actually happens.
+* How often reproduces?: what percentage of the time does it reproduce?
+* Version: the version of the app.
+* Repository: Link to the repository you are working with.
+* Operating system: The operating system used.
+* Additional information: Any additional to help to reproduce. (screenshots, animated gifs)
+
+## Pull Requests
+
+1. Fork the project
+2. Implement feature/fix bug & add test cases
+3. Ensure test cases & static analysis runs succesfully
+4. Submit a pull request to `master` branch
+
+Please include unit tests where necessary to cover any functionality that is introduced.
+
+## Coding Guidelines
+
+* All features or bug fixes **must be tested** by one or more unit tests/specs
+* All public API methods **must be documented** and potentially also described in the user guide.
+* All Objective-C code should follow the guidelines of the existing code.
+
+## Commit messages
+
+Each commit message consists of a header, a body and a footer. The header has a special format that includes a type, a scope and a subject:
+
+```
+<type>(<scope>): <subject>
```

Comment:
Nah, it should be removed here too.

Reply:
> Nah, it should be removed here too.

On second thoughts I'd like to leave it in as optional because it could refer to a specific part of the module e.g. networkApiClient, configFetcher, configParser etc.